src/lib/structures/Command.ts

```diff
@@ -248,6 +248,11 @@ export class Command<PreParseReturn = Args, O extends Command.Options = Command.
 	}
 
 	public override async reload() {
+		// If this command does not have any application commands, just call the super method
+		if (!this.supportsChatInputCommands() && !this.supportsContextMenuCommands()) {
```

Comment:
Instead, you should place this logic on line 301 of the current diff (297 originally), and check if the registry got any new entries